.script/solutionValidator.ts

```diff
@@ -0,0 +1,128 @@
+import { runCheckOverChangedFiles } from "./utils/changedFilesValidator";
+import * as logger from "./utils/logger";
+import { ExitCode } from "./utils/exitCode";
+import fs from "fs";
+import { MainTemplateValidationError } from "./utils/validationError";
+
+// initialize arrays to store valid domains and verticals
+let validDomains: string[] = [];
+let validVerticals: string[] = [];
+
+// read the valid domains and verticals from the JSON file
+try {
+    const validDomainsVerticals = JSON.parse(fs.readFileSync('./.script/ValidDomainsVerticals.json', "utf8"));
+    validDomains = validDomainsVerticals.validDomains;
+    validVerticals = validDomainsVerticals.validVerticals;
+} catch (error) {
+    logger.logError(`Error reading ValidDomainsVerticals.json file: ${error}`);
+}
+
+// function to check if the solution is valid
+export async function IsValidSolution(filePath: string): Promise<ExitCode> {
+
+    // check if the file is a mainTemplate.json file
+    if (filePath.endsWith("mainTemplate.json")) {
+        // read the content of the file
+        let jsonFile = JSON.parse(fs.readFileSync(filePath, "utf8"));
+
+        // check if the file has a "resources" field
+        if (!jsonFile.hasOwnProperty("resources")) {
+            console.warn(`No "resources" field found in the file. Skipping file path: ${filePath}`);
+            return ExitCode.SUCCESS;
+        }
+
+        // get the resources from the file
+        let resources = jsonFile.resources;
+
+        // filter resources that have type "Microsoft.OperationalInsights/workspaces/providers/metadata"
+        const filteredResource = resources.filter(function (resource: { type: string; }) {
+            return resource.type === "Microsoft.OperationalInsights/workspaces/providers/metadata";
+        });
+        if (filteredResource.length > 0) {
+            filteredResource.forEach((element: { hasOwnProperty: (arg0: string) => boolean; properties: { hasOwnProperty: (arg0: string) => boolean; categories: any; }; }) => {
+                // check if the resource has a "properties" field
+                if (element.hasOwnProperty("properties") === true) {
+                    // check if the "properties" field has a "categories" field
+                    if (element.properties.hasOwnProperty("categories") === true) {
+                        const categories = element.properties.categories;
+
+                        let invalidDomains = [];
+                        let invalidVerticals = [];
+
+                        // check if the categories have a "domains" field
+                        if (categories.hasOwnProperty("domains")) {
```

Comment:
We need to fail in case either "domains" property itself is not available or don't have at least 1 value.

Reply:
Added logic for at least 1 domain